core/Int.carp

```diff
@@ -1,72 +1,10 @@
 (system-include "carp_int.h")
 
 (defmodule Int
-  (register MAX Int "CARP_INT_MAX")
-  (register MIN Int "CARP_INT_MIN")
-  (register + (λ [Int Int] Int))
-  (register - (λ [Int Int] Int))
-  (register * (λ [Int Int] Int))
-  (register / (λ [Int Int] Int))
-  (register < (λ [Int Int] Bool))
-  (register > (λ [Int Int] Bool))
-  (register = (λ [Int Int] Bool))
-  (register neg (λ [Int] Int))
-  (register mod (λ [Int Int] Int))
-  (register bit-shift-left (λ [Int Int] Int))
-  (register bit-shift-right (λ [Int Int] Int))
-  (register bit-and (λ [Int Int] Int))
-  (register bit-or (λ [Int Int] Int))
-  (register bit-xor (λ [Int Int] Int))
-  (register bit-not (λ [Int] Int))
-  (register inc (λ [Int] Int))
-  (register dec (λ [Int] Int))
-  (register copy (λ [&Int] Int))
-
-  (doc abs "The absolute value (removes the negative sign) of an Int.")
-  (register abs (λ [Int] Int))
-
-  (defn even? [a] (= (mod a 2) 0))
-  (defn odd? [a] (not (even? a)))
-
+  (Numeric.register Int)
+  (Integral.register Int)
   (defn zero []
     0)
-
-  (defn add-ref [x y]
-    (Int.+ @x @y))
-
-  ;; Move to generic math module?
-  (defn clamp [min, max, val]
-    (if (> val max)
-      max
-      (if (< val min)
-        min
-        val)))
-
-  (doc pow "Raise x to the power of y.")
-  (defn pow [x y]
-    (let-do [r 1]
-      (while (/= y 0)
-        (do
-          (when (/= (bit-and y 1) 0)
-            (set! r (* r x)))
-          (set! y (/ y 2))
-          (set! x (* x x))))
-      r))
-
-  (doc positive-mod "Like mod but always returns a positive answer.")
-  (defn positive-mod [k n]
-    (let [r (Int.mod k n)]
-      (if (> 0 r)
-        (+ r n)
-        r)))
-
-  (doc to-int "acts as the identity function to implement the interface.")
-  (sig to-int (Fn [Int] Int))
-  (defn to-int [a] a)
-
-  (doc from-int "acts as the identity function to implement the interface.")
-  (sig from-int (Fn [Int] Int))
-  (defn from-int [a] a)
 )
 
 (defmodule IntRef
```

Comment:
The Integral.register macro could maybe generate these functions too?

Reply:
Do you mean moving the `IntRef` functions to the `Int` module?

Reply:
Addressed this explicitly.